heron/scheduler-core/tests/java/com/twitter/heron/scheduler/SubmitterMainTest.java

```diff
@@ -124,21 +140,33 @@ public void testSubmitTopology() throws Exception {
     when(config.getStringValue(ConfigKeys.get(UPLOADER_CLASS))).thenReturn(UPLOADER_CLASS);
 
     // Failed to validate the submission
-    doReturn(false).when(submitterMain)
+    doThrow(new TopologySubmissionException(
+        String.format("Topology %s already exist", topology.getName()))).when(submitterMain)
         .validateSubmit(any(SchedulerStateManagerAdaptor.class), anyString());
-    assertFalse(submitterMain.submitTopology());
+
+    try {
+      submitterMain.submitTopology();
+      fail("submitTopology should throw exception");
+    } catch (TopologySubmissionException e) {
+      assertEquals(String.format("Topology %s already exist", topology.getName()), e.getMessage());
```

Comment:
I am testing if the exception will be thrown out. If no exception is thrown out, the control flow reaches `fail`, which will fail the test.

And this `SubmitterMainTest` shows the problem I mentioned days ago. Inside the function `testSubmitTopology`, we are testing multiple statements to see if they throw exceptions. If we want to use `Test(expected=xxxException.class)`, we need to split this `testSubmitTopology` into multiple sub-tests (and duplicate the code that sets up the test).